rustls/src/server/hs.rs

```diff
@@ -265,6 +253,26 @@ impl ExtensionProcessing {
         Ok(())
     }
 
+    fn set_sct(
+        &mut self,
+        can_use: bool,
+        sess: &ServerSessionImpl,
+        server_key: &mut sign::CertifiedKey
+    ) {
+        if let Some(list) = server_key.take_sct_list() {
+            if can_use {
+                self.send_sct = true;
+
+                if !sess.common.is_tls13() {
+                    self.exts.push(ServerExtension::make_sct(list));
+                    return;
+                }
+            }
+            // Restore list if we were not allowed to use it
+            server_key.sct_list = Some(list);
```

Comment:
I'm not sure what the separate function is buying us here (just the early return?) and I'd like it to be more obvious that a taken-but-unused sct list is always put back. Would you like to do another take?

Reply:
I moved it into a function so I could use the early return in combination with the fallback list reset. I tried keeping it inline but I was not able to find a clean way to set the `send_sct` flag with the same rules as before without either cloning `server_key.sct_list` or having multiple places the list would need put back.

Is the list generally large enough that cloning it would be expensive? That would be cleaner then whats presented here.

Reply:
I would like to see an attempt that avoids factoring out the function. Having the separate function makes it harder to understand the code at a higher level because we have to search for callers to that function to see that there is only this one.

Reply:
I'm pretty sure there's a way to do this without cloning.

Reply:
PTAL at the new attempt. The new function has been removed and its now more clear when the taken list is put back.